crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -157,9 +158,22 @@ impl SyncTaskSupervisor {
 
         let wait_to_be_online = async move {
             loop {
+                // We're in an infinite loop, but our request sending already has an exponential
+                // backoff set up. This will kick in for any request errors that we consider to
+                // be transient. Common network errors (timeouts, DNS failures)
+                // or any server error in the 5xx range of HTTP errors are
+                // considered to be transient.
+                //
+                // This means that we're not going to tightloop here, and in the case the
+                // `RequestConfig` has been set up to not have a limit, ever hit the second
+                // iteration of this loop.
+                //
+                // No matter, as a precaution, we're going to sleep for a while here in the
+                // Error case, the user might have configured the RequestConfig
+                // to not contain any backoff or retries.
```

Comment:
Doesn't it defeat the purpose of exponential backoff, though? The first retries would happen after less than 500ms, with exponential backoff, so we're artificially delaying the restart here.

Reply:
Same argument as https://github.com/matrix-org/matrix-rust-sdk/pull/4592#discussion_r1936844342.